File: LeetCode/1390/main.cpp
class Solution {
public:
  int check(int n) {
    int c = 0, s = 0;
    for (int i = 1; i * i <= n; i++) {
      if (n % i == 0) {
        if (n / i == i) {
          return 0;
        }
        c += 2;
        s += i;
        s += (n / i);
      }
    }
    return c == 4 ? s : 0;
  }

  int sumFourDivisors(vector<int> &arr) {
    int n = arr.size();
    int ans = 0;

    for (int i = 0; i < n; i++) {
      ans += check(arr[i]);
    }

    return ans;
  }
};

File: LeetCode/1390/main.py
class Solution:
    def sumFourDivisors(self, arr: List[int]) -> int:

        def check(n):
            c, s = 0, 0
            for i in range(1, int(n**0.5) + 1, 1):
                if (n % i == 0):
                    if (i * i == n):
                        return 0

                    c += 2
                    s += (i + (n // i))

            return s if c == 4 else 0

        ans = 0

        for x in arr:
            ans += check(x)

        return ans
